app/services/saml_request_validator.rb

```diff
@@ -32,13 +32,37 @@ def authorized_service_provider
   end
 
   def authorized_authn_context
-    if !Saml::Idp::Constants::VALID_AUTHN_CONTEXTS.include?(authn_context) ||
-       (Saml::Idp::Constants::IAL2_AUTHN_CONTEXTS.include?(authn_context) &&
-            service_provider.ial != 2)
+    if !valid_authn_context? ||
+       (ial2_context_requested? && service_provider.ial != 2)
       errors.add(:authn_context, :unauthorized_authn_context)
     end
   end
 
+  def valid_authn_context?
+    case authn_context
+    when Array
+      authn_contexts = authn_context.reject do |classref|
+        %r{#{::Regexp.quote(Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF)}}.match? classref
+      end
+      authn_contexts.all? do |classref|
+        Saml::Idp::Constants::VALID_AUTHN_CONTEXTS.include?(classref)
+      end
+    else
+      Saml::Idp::Constants::VALID_AUTHN_CONTEXTS.include?(authn_context)
+    end
+  end
+
+  def ial2_context_requested?
+    case authn_context
+    when Array
```

Comment:
ya can we just make this `authn_context` always an array?

Reply:
Gah!

Reply:
Just saw the lastest commit ... since sometimes it can be a single element, let's just cast it:

```
Array(authn_context)
```

will do the right thing and always give us an array and not need to be nested

Reply:
Got it -- thanks!